tools/mo/openvino/tools/mo/middle/SSliceComplex.py

```diff
@@ -109,7 +109,10 @@ def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
 
         for dst in complex_node.out_port(0).get_connection().get_destinations():
             after_complex_node = dst.node
-            after_complex_node['input_rank_changed'] = True
+            if after_complex_node.type == 'Roll':
+                add_constant_to_negative_values(after_complex_node, 2, int64_array(emulated_complex_tensor_rank))
+            else:
+                after_complex_node['input_rank_changed'] = True
```

Comment:
Just to ensure, `after_complex_node['input_rank_changed'] = True` is not needed when the node type is `Roll`?

Reply:
'input_rank_changed` is used by other transformations to adjust op after rank changing and then the key is removed. For 'Roll' we already adjust it so 'input_rank_changed` seems to be not needed

Reply:
Now I see that `input_rank_changed` is used only by the front transformation `CorrectRollAxes` that goes earlier in the transformation pipeline.
Now this attribute is useless. Please tell me -why do we need it then?

Reply:
In any case we need to have debug info in other places to mark-up all operations affected by complex numbers. Unfortunately, we don't have this mechanism now. In our plans we have to implement the support of complex numbers in our opset and in the frontend and try to avoid from complex number in common transformation if plugin does not support it. In this solution we will see all edges with complex numbers and try to replace it with float emulation.